doc/site/content/en/docs/Contribution guidelines/dev_new_stages.md

```diff
@@ -7,7 +7,93 @@ categories: [Examples]
+```
+super(config, new StageSpec()
+  .withRequiredProperties("<requiredProperty>")
+  .withOptionalProperties("<optionalProperty>")
+  .withRequiredParents("<requiredParent>"));
+```
+5. Define instance variables that correlate to config properties you wish to request from the user (both required and optional)
+   - The following code shows examples of common patterns used to extract config parameters; reference the Config code for more methods
+
+```config.getConfig("<nameOfProperty>").root().unwrapped(); // for required properties```
+
+```config.hasPath("<nameOfProperty>") ? config.getInt("<nameOfProperty>") : <defaultValue>;```
+
+```ConfigUtils.getOrDefault(config, "<nameOfProperty>", <defaultValue>;```
+
+6. Add the abstract method processDocument to your class
+   - This method is where we want to make changes to fields on the document and potentially create child documents 
+   - This method should return null assuming we are not intending to generate child documents
+     - Reference the Javadoc in the Stage class for more information on how to support this functionality
+
+7. Add appropriate comments to explain any important code in the class and add Javadoc before the class declaration
+   - Javadoc should explain the behaviour of this Stage and also should list config parameters, their types, whether they are optional, and a short description
+
+## Unit Testing:
+Lucille uses JUnit as its testing framework, please refer to JUnit best practices when making tests. 
+
+1. Create a new Java class underneath the following directory in Lucille; name should be the same as the Stage’s name, with ```Test``` appended to the end
+
+   - ```lucille/lucille-core/src/test/java/com/kmw/lucille/stage/```
+2. Create a new directory underneath the following directory in Lucille: name should same as the testing class' name
+
+   - ```lucille/lucille-core/src/test/resources/```
+
+   - Underneath this directory create a new file called config.conf; this will be an example config that will be used in our test class; you can create more for further testing
+
+3. The following code snippet can be used to create a new Stage with the provided config name
+
+```StageFactory.of(<StageName>.class).get("<StageTestName>/config.conf");```
+
+4. The following code snippet will process a given Document; reference the Document class for more information
+```s.processDocument(d); // where s is the Stage and d is the Document```
+ 
+
+The following are standards for testing in Lucille:
+
+- There should be at least one unit test for a Stage
+
+- Recommended to aim for 80% code coverage, but not a hard cutoff
+
+- When testing services, use mocks and spys
+
+- Tests should test notable exceptions thrown by Stage class code
+
+- Tests should cover all logical aspects of a Stage’s function
+
+- Tests should be deterministic
+
+- Code coverage should not only encapsulate aspects of lucille-core but also modules; reference Lucille documentation for more info on this

```

Comment:
Should remove "; reference Lucille documentation for more info on this" since we don't have any documentation yet on creating unit tests for modules

Reply:
done in [7042081](https://github.com/kmwtechnology/lucille/pull/200/commits/7042081cfc6c1fdc2945f0fd70c127e074b7e835)